### ISSUE TITLE
operator resources not updating

### ISSUE THREAD

**Issue by jurgen-weber-deltatre** [mongodb/mongodb-kubernetes-operator#1234]
**What did you do to encounter the bug?**

Update to CRD Community resource 

```
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
```

Does not start stateful set rolling update.

**What did you expect?**
I expect the operator to reconcile its resources and update them.

**What happened instead?**
I can see in the logs the operator has reconciled the new STS resource with my changes, but it has not updated it on the cluster.

**Screenshots**
```
mongodb-kubernetes-operator-57c588ddf9-t77xq mongodb-kubernetes-operator 2023-03-01T05:55:35.838550296+11:00 2023-02-28T18:55:35.838Z	INFO	controllers/replica_set_controller.go:265	Successfully finished reconciliation, MongoDB.Spec: {Members:3 Type:ReplicaSet Version:4.2.23 Arbiters:0 FeatureCompatibilityVersion: ReplicaSetHorizons:[] Security:{Authentication:{Modes:[SCRAM] IgnoreUnknownUsers:0xc001f39160} TLS:{Enabled:true Optional:false CertificateKeySecret:{Name:account-mongodb-cert-manager-tls-certificate} CaCertificateSecret:&LocalObjectReference{Name:account-mongodb-tls-selfsigned-ca,} CaConfigMap:nil} Roles:[]} Users:[{Name:deltatrescale DB:admin PasswordSecretRef:{Name:account-mongodb-deltatrescale Key:} Roles:[{DB:admin Name:clusterAdmin} {DB:admin Name:userAdminAnyDatabase}] ScramCredentialsSecretName:account-mongodb-deltatrescale ConnectionStringSecretName:}] StatefulSetConfiguration:{SpecWrapper:{Spec:{Replicas:<nil> Selector:nil Template:{ObjectMeta:{Name: GenerateName: Namespace: SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[traffic.sidecar.istio.io/excludeInboundPorts:27017] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[]} Spec:{Volumes:[] InitContainers:[] Containers:[] EphemeralContainers:[] RestartPolicy: TerminationGracePeriodSeconds:<nil> ActiveDeadlineSeconds:<nil> DNSPolicy: NodeSelector:map[] ServiceAccountName: DeprecatedServiceAccount: AutomountServiceAccountToken:<nil> NodeName: HostNetwork:false HostPID:false HostIPC:false ShareProcessNamespace:<nil> SecurityContext:nil ImagePullSecrets:[] Hostname: Subdomain: Affinity:nil SchedulerName: Tolerations:[] HostAliases:[] PriorityClassName: Priority:<nil> DNSConfig:nil ReadinessGates:[] RuntimeClassName:<nil> EnableServiceLinks:<nil> PreemptionPolicy:<nil> Overhead:map[] TopologySpreadConstraints:[] SetHostnameAsFQDN:<nil> OS:nil}} VolumeClaimTemplates:[] ServiceName: PodManagementPolicy: UpdateStrategy:{Type: RollingUpdate:nil} RevisionHistoryLimit:<nil> MinReadySeconds:0 PersistentVolumeClaimRetentionPolicy:nil}}} AdditionalMongodConfig:{Object:map[]} AutomationConfigOverride:<nil> Prometheus:<nil>}, MongoDB.Status: {MongoURI:mongodb://account-mongodb-0.account-mongodb-svc.account.svc.cluster.local:27017,account-mongodb-1.account-mongodb-svc.account.svc.cluster.local:27017,account-mongodb-2.account-mongodb-svc.account.svc.cluster.local:27017/?replicaSet=account-mongodb Phase:Running Version:4.2.23 CurrentStatefulSetReplicas:3 CurrentMongoDBMembers:3 CurrentStatefulSetArbitersReplicas:0 CurrentMongoDBArbiters:0 Message:}	{"ReplicaSet": "account/account-mongodb"}
```

```
$ kc get sts account-mongodb -o yaml  | grep istio
        sidecar.istio.io/inject: "false"
        traffic.sidecar.istio.io/excludeInboundPorts: "27017"
        traffic.sidecar.istio.io/excludeOutboundPorts: "27017"
```
The first output is from the operator, and you can see it has reconciled the new STS and you will note under Annotations there is only ONE istio annotation...

The second output is me just grepping the STS resource and you can see, all of the istio sidecar annotations I had are still present.. So it is not applying the update.   
        

**Operator Information**
 - Operator Version 0.7.8
 - MongoDB Image used, default image Version:4.2.23.

**Kubernetes Cluster Information**
AWS EKS
```
Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.1", GitCommit:"8f94681cd294aa8cfd3407b8191f6c70214973a4", GitTreeState:"clean", BuildDate:"2023-01-18T15:51:24Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"24+", GitVersion:"v1.24.8-eks-ffeb93d", GitCommit:"abb98ec0631dfe573ec5eae40dc48fd8f2017424", GitTreeState:"clean", BuildDate:"2022-11-29T18:45:03Z", GoVersion:"go1.18.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**

Full resource

```
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  annotations:
    generation: 2
  labels:
    app: raw
    chart: raw-0.2.4
    heritage: Helm
    release: account-mongodb
  name: account-mongodb
  namespace: account
spec:
  members: 3
  security:
    authentication:
      ignoreUnknownUsers: true
      modes:
        - SCRAM
    tls:
      caCertificateSecretRef:
        name: account-mongodb-tls-selfsigned-ca
      certificateKeySecretRef:
        name: account-mongodb-cert-manager-tls-certificate
      enabled: true
  statefulSet:
    spec:
      template:
        metadata:
          annotations:
            traffic.sidecar.istio.io/excludeInboundPorts: '27017'
  type: ReplicaSet
  users:
    - db: admin
      name: deltatrescale
      passwordSecretRef:
        name: account-mongodb-deltatrescale
      roles:
        - db: admin
          name: clusterAdmin
        - db: admin
          name: userAdminAnyDatabase
      scramCredentialsSecretName: account-mongodb-deltatrescale
  version: 4.2.23
status:
  currentMongoDBMembers: 3
  currentStatefulSetReplicas: 3
  mongoUri: >-
    mongodb://account-mongodb-0.account-mongodb-svc.account.svc.cluster.local:27017,account-mongodb-1.account-mongodb-svc.account.svc.cluster.local:27017,account-mongodb-2.account-mongodb-svc.account.svc.cluster.local:27017/?replicaSet=account-mongodb
  phase: Running
  version: 4.2.23
```

**Comment by jurgen-weber-deltatre** [comment]
after some more experimentation, it seems if I update teh spec it will update the resource and the rolling update happens, but if you just change an annotation (which I would expect to have the same behaviour) it does not.

**Comment by dan-mckean** [comment]
Hi, I'm Dan and I'm the Product Manager for MongoDB's support of Kubernetes.

I'm doing some work right now to try and identify how the Community Operator is being used. The Community Operator is something I inherited when I started at MongoDB, but it doesn't get as much attention from us as we'd like and we're trying to understand how it's used in order to establish its future. It will help us prioritize future issues and PRs raised by the community 🙂

Here's a super short survey (it's much easier for us to review all the feedback that way!): [https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link](https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link)

 If you'd rather email me instead: [dan.mckean@mongodb.com](mailto:dan.mckean@mongodb.com?subject=MongoDB%20Community%20Operator%20feedback)

Thank you in advance!
Dan
